pkg/redirectpolicy/manager.go

```diff
@@ -396,23 +389,15 @@ func (rpm *Manager) updateConfigSvcFrontend(config *LRPConfig, frontends ...*fro
 	rpm.policyConfigs[config.id] = config
 }
 
-func (rpm *Manager) filterBackends(fe *feMapping, backends ...backend) []backend {
-	var newBackends []backend
-	for _, currBk := range fe.backends {
-		for _, removeBk := range backends {
-			if removeBk.StringWithProtocol() != currBk.StringWithProtocol() {
-				newBackends = append(newBackends, currBk)
+func (rpm *Manager) deletePolicyBackends(config *LRPConfig, podID podID) {
+	for _, fe := range config.frontendMappings {
+		for i, be := range fe.podBackends {
+			// Remove the pod from the frontend's backends slice.
+			if be.podID == podID {
+				fe.podBackends[i] = fe.podBackends[len(fe.podBackends)-1]
+				fe.podBackends = fe.podBackends[:len(fe.podBackends)-1]
```

Comment:
I would have found a little bit surprising the order of backends to be changed if I was deleting them. I would suggest to add this caveat in the function description or keep the order on deletion.

Also, needs unit tests.

Reply:
The load balancing framework does round-robin during service translation anyway so the order is not super critical, is it? I'll keep the order same nevertheless.

Reply:
Probably @brb would be the one to give you a better idea of whether ordering matters. I wouldn't be surprised if it plays into the "service stickiness" implementation, but I don't actually know about this off the top of my head.